### DISCUSSION TITLE
Merge Window Throw and Quick Throw

### DISCUSSION THREAD

**Discussion by adamshackleford** [rxhanson/RectanglePro-Community#50] [Ideas]
I wrote the message below before I discovered the overlap between Window Throw and Quick Throw. I can do what I want with some Quick Throws, but it feels weird having two nearly-identical features.

Can these be merged into one feature?

Window Throw's pre-throw visualization is so slick, whereas Quick Throw is unrefined by comparison.
I guess I'm really asking for Window Throw to be enhanced with more modifiers, and Quick Throw to be removed.
* * *
Window Throw is one of my favorite Rectangle Pro features.

It's less mental load (for me) to remember Window Throws than a bunch of Quick Throws. Maybe it's because Window Throw has 8 "arrow keys" whereas Quick Throw has 4.

I'm trying to get > 16 throw actions (want sixths and center-half and more..., and don't want to give up the defaults).

At least a couple of ways to go here:

1. Have *two* (or N, I guess) triggers, so instead of 16 throw actions, I could have 32, or 48, or... So we might have Cmd+Ctrl and Shift+Cmd+Ctrl (configurable obviously).

2. Have there-and-back throw gestures. Instead of (for example) moving the mouse pointer North, move it North then immediately South.

**Comment by rxhanson** [comment]
Quick Throw is its own feature, and I would have no desire/plans to remove what currently exists. 

In my opinion, what you're looking for is outside of the realm of what most users would look for in the Window Throw. There's a certain degree where the added complexity might start to degrade the approachability of this functionality. When I initially created the Window Throw, before adding the Long Throw, I was pretty set on having a list view with unlimited triggers, but after a lot of thought I felt like it was hard to justify implementing it this way when most users would probably just want one and the list view would be unapproachable. I've already seen this with how much people avoid using the Custom Shortcuts tab. From a user standpoint, more triggers means more things to memorize, too. I could definitely add another tab for more window throw triggers with the list view, but I'm not convinced that this added functionality would be desirable for most users, and there's a point where additional features have to really be scrutinized to determine if they add enough value to the app for enough people to justify additional UI complexity, code complexity, etc. I prefer simplicity here, despite there being a decent amount of complexity already in the app. Obviously, if there are a ton of people that want this functionality, then I can see adding this in. 

As an alternative solution, I have plans to add in some form of a Fancy Zones implementation, where you can set up your own zones and do something like a window throw except the action will be relative to the preconfigured rectangular zone that the cursor is in, rather than the 8 slices around the reticle point. I think this will be more in line with a concept of having a lot different specific window positions in mind, and it should be pretty straightforward to have the list view with infinite triggers based on things I have already built into the app. Going this route instead of adding more Window Throw triggers would also just keep that feature simpler, and I have incentive to keep it that way. Also, there's a lot of people that miss Fancy Zones on Windows, creating a larger incentive to build something akin to that functionality.

**Reply by adamshackleford** [reply]
Thanks for your excellent answer. 

Definitely, more triggers == more complexity. OTOH, if a user is using both features, they're already dealing with not only two triggers, but two *features*. You could argue that:

complexity(Window Throw w/two triggers) < complexity(Window Throw w/one trigger + Quick Throw w/one trigger)

Your Settings UI would lose widgets, I think, if Window Throw got a little fancier, and Quick Throw went to meet its ancestors.

I had to Google "Fancy Zones". Pretty slick. PowerToys has gotten a lot beefier since "Ctrl2Cap".

Anyway, Rectangle Pro is great. I ponied up for a license. Don't spend all the money in one place :)

**Reply by rxhanson** [reply]
I'm not here to argue, but Quick Throw is my preferred way to use Rectangle Pro. This could very well be a feature that is not highly used, and I could absorb it into Window throw, but it's quite honestly the one feature I really want. If I ever end up being forced to use another OS for a job, I would build the Quick Throw as an app and just use that.

Thanks for purchasing! I enjoy the feedback.

**Reply by adamshackleford** [reply]
OK, I'm going to try going all Quick Throw. 

(a few days later...)

New discussion in Ideas: "Eliminate the superfluous Window Throw feature"
